src/main.rs

```diff
@@ -325,3 +475,77 @@ async fn orchestrator_post_request(
     tracing::debug!("Parsed JSON response: {:?}", json);
     Ok(serde_json::from_value(json).expect("unexpected json response from request"))
 }
+
+async fn orchestrator_streaming_request(
+    payload: Option<&mut Map<String, Value>>,
+    headers: &HeaderMap,
+    url: &str,
+    client: &reqwest::Client,
+) -> Result<impl futures::Stream<Item = Result<String, anyhow::Error>>, anyhow::Error> {
+    tracing::debug!(
+        "Sending streaming POST request to {} with payload: {:?}",
+        url,
+        payload
+    );
+
+    let mut req = client.post(url).json(&payload);
+
+    // Forward authorization headers
+    for (name, value) in headers.iter() {
+        tracing::debug!("Header {}: {:?}", name, value);
+        let name_str = name.as_str().to_ascii_lowercase();
+        if name_str == "authorization" {
+            req = req.header(name, value);
+        }
+        if name_str.starts_with("x-forwarded") {
+            req = req.header(name, value);
+        }
+    }
+
+    let response = req.send().await.map_err(|e| {
+        tracing::error!("Failed to send streaming request or connect to orchestrator: {:?}", e);
+        anyhow::anyhow!("Failed to send streaming request or connect to orchestrator: {:?}", e)
+    })?;
```

Comment:
This seems a bit redundant, maybe you can use anyhow context:
```suggestion
    let response = req
    .send()
    .await
    .context("Failed to send streaming request or connect to orchestrator")?;
```